holmes/core/truncation/compaction.py

```
@@ -86,6 +86,75 @@ def _strip_images_for_compaction(messages: list[dict]) -> list[dict]:
     return stripped
 
 
+def _append_text_to_content(content: Any, text: str) -> Any:
+    """Append a text snippet to a message content (str, block list, or None)."""
+    if not content:
+        return text
+    if isinstance(content, str):
+        return f"{content}\n{text}"
+    if isinstance(content, list):
+        return content + [{"type": "text", "text": text}]
```

Ruff (0.15.17)

[warning] Consider `[*content, {"type": "text", "text": text}]` instead of concatenation

Replace with `[*content, {"type": "text", "text": text}]`

(RUF005)